packages/authenticator/amplify_authenticator/lib/src/widgets/social/social_button.dart

```diff
@@ -81,24 +83,34 @@ class SocialSignInButton extends AuthenticatorButton<SocialSignInButton> {
   const SocialSignInButton({
     super.key,
     required this.provider,
+    required Widget this.customIcon,
```

Comment:
It's not really custom since it's just a required attribute for rendering the button. Also thinking now it should be optional.

```suggestion
    required Widget this.icon,
```